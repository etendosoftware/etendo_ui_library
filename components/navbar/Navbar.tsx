import {Dimensions, View} from 'react-native';
import React from 'react';
import {styles} from './Navbar.styles';
import {NavbarProps} from './Navbar.types';
import {Welcome, Profile, EtendoLogo} from './index';
import MenuBurger from './components/MenuBurger/MenuBurger';

const Navbar = ({
  name,
  email,
  leftComponent,
  rightComponent,
  optionsProfile,
  profileImage,
  onOptionSelectedProfile,
  onPressMenuBurger,
  onPressLogo,
}: NavbarProps) => {
  const screenWidth = Dimensions.get('window').width;
  const isTablet = screenWidth > 700;

  return (
    <View style={styles.container}>
      <View style={styles.leftContainer}>
        {isTablet ? (
          <>
            <EtendoLogo onPress={onPressLogo} />
            <Welcome name={name + '!'} />
            {leftComponent}
          </>
        ) : (
          <MenuBurger onPress={onPressMenuBurger}></MenuBurger>
        )}
      </View>
      <View style={styles.rightContainer}>
        {isTablet ? <>{rightComponent}</> : null}
        <Profile
          data={optionsProfile}
          name={name}
          email={email}
          onOptionSelected={onOptionSelectedProfile}
          profileImage={profileImage}
        />
      </View>
    </View>
  );
};

export default Navbar;
