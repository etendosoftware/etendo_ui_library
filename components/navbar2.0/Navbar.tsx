import {Alert, Dimensions, TouchableOpacity, TouchableWithoutFeedback, View} from 'react-native';
import React, {useState} from 'react';
import {styles} from './Navbar.styles';
import {NavbarProps} from './Navbar.types';
import Welcome from './components/welcome/Welcome';
import Logo from './components/logo/Logo';

import Profile from './components/profile/Profile';
import Notification from './components/notification/Notification';
const height = Dimensions.get('window').height;
const width = Dimensions.get('window').width;

const Navbar = ({name,email,leftComponent, rightComponent,anyNotifications, optionsNotifications,onOptionSelectedNotifications,onViewAllNotifications,onMarkAllAsReadNotifications}: NavbarProps) => {

  return (
    <View style={{flex: 1}}>
      <View style={styles.container}>
        <View style={styles.leftContainer}>
          <Logo onPress={()=>{}}/>
          <Welcome name={name + '!'}/>
          {leftComponent}
        </View>
        <View style={styles.rightContainer}>
          {rightComponent}
          <Notification
           anyNotification={anyNotifications}
           optionsNotifications={optionsNotifications}
           onOptionSelectedNotifications={onOptionSelectedNotifications} 
           onViewAllNotifications={onViewAllNotifications}
           onMarkAllAsReadNotifications={onMarkAllAsReadNotifications}
          />
          <Profile name={name} email={email}/>
        </View>
      </View>
    </View>
  );
};

export default Navbar;
