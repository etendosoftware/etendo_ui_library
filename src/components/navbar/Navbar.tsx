import { Dimensions, View } from 'react-native';
import React, { useState } from 'react';
import { marginIncomponents, styles } from './Navbar.styles';
import { NavbarProps, OptionProfileItem, RightComponent } from './Navbar.types';
import { Welcome, Profile, EtendoLogo } from './index';
import MenuBurger from './components/MenuBurger/MenuBurger';
import { isTablet } from '../../helpers/table_utils';

const Navbar = ({
  name,
  title,
  email,
  rightComponent,
  optionsProfile,
  profileImage,
  endOptions,
  onOptionSelectedProfile,
  onPressMenuBurger,
  onPressLogo,
}: NavbarProps) => {
  const width = Dimensions.get('window').width;
  const widthLeft = styles.leftContainer.width;
  const profileWidth = isTablet() ? 84 : 72;

  const [renderedComponents, setRenderedComponents] = useState<
    RightComponent[] | undefined
  >(rightComponent);

  const [removedComponents, setRemovedComponents] = useState<
    OptionProfileItem[] | undefined
  >([]);

  const [isLoading, setIsLoading] = useState<boolean>(true);

  let componentWidths = new Array(rightComponent?.length).fill(0);

  const calculateRenderedComponents = (index: number, newWidth: number) => {
    componentWidths[index] = newWidth;
    if (rightComponent) {
      if (index === rightComponent?.length - 1) {
        let totalWidth = componentWidths.reduce(
          (total, width) => total + width,
          0,
        );

        while (
          totalWidth + widthLeft + profileWidth - marginIncomponents >
          width
        ) {
          const lastComponentIndex = componentWidths.length - 1;

          if (renderedComponents?.[lastComponentIndex]?.inOptions) {
            setRemovedComponents(prevRemovedComponent => {
              const updatedRemovedComponent = [
                renderedComponents[lastComponentIndex].inOptions,
                ...(prevRemovedComponent ?? []),
              ];
              return updatedRemovedComponent as OptionProfileItem[];
            });
          }

          totalWidth -= componentWidths[lastComponentIndex];
          componentWidths.splice(lastComponentIndex, 1);
        }

        setRenderedComponents(prev => {
          const slicedComponents = prev?.slice(0, componentWidths.length);
          return slicedComponents?.reverse();
        });
        setIsLoading(false);
      }
    }
  };

  return (
    <View style={styles.container}>
      <View style={[styles.leftContainer]}>
        {isTablet() ? (
          <>
            <EtendoLogo onPress={onPressLogo} />
            <Welcome name={name} title={title} />
          </>
        ) : (
          <MenuBurger onPress={onPressMenuBurger} />
        )}
      </View>
      <View style={[styles.rightContainer]}>
        <>
          {renderedComponents?.map((item, index) => (
            <View
              key={index}
              onLayout={event => {
                const { width } = event.nativeEvent.layout;
                calculateRenderedComponents(index, width + marginIncomponents);
              }}
              style={[styles.itemMapContainer, isLoading && { opacity: 0 }]}>
              {item.component}
            </View>
          ))}
        </>
        <View>
          <Profile
            profileOptions={optionsProfile}
            otherOptions={removedComponents}
            name={name}
            email={email}
            onOptionSelected={onOptionSelectedProfile}
            profileImage={profileImage}
            endOptions={endOptions}
          />
        </View>
      </View>
    </View>
  );
};

export default Navbar;
