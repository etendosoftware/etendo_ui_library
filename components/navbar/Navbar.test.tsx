import React from 'react';
import renderer from 'react-test-renderer';
import {
  binocularIcon,
  chatIcon,
  etendoLogo,
  hamburgerIcon,
  slideShow,
  userNoBorder,
} from '../../assets/images/icons';

import {Navbar} from './Navbar';

const datasetNavbarH = [
  {
    routeImage: etendoLogo,
    routeNav: '',
    key: 'dataset0',
    name: 'FirstName LastName',
  },
  {
    routeImage: hamburgerIcon,
    routeNav: '',
    key: 'dataset1',
    name: 'FirstName LastName',
  },
  {
    routeImage: binocularIcon,
    routeNav: '',
    key: 'dataset2',
    name: 'FirstName LastName',
  },
  {
    routeImage: slideShow,
    routeNav: '',
    key: 'dataset3',
    name: 'FirstName LastName',
  },
  {
    routeImage: chatIcon,
    routeNav: '',
    key: 'dataset4',
    name: 'FirstName LastName',
  },
  {
    routeImage: userNoBorder,
    routeNav: '',
    key: 'dataset5',
    name: 'FirstName LastName',
  },
];

const onCheckSelectedMock = () => datasetNavbarH[0];

describe('Running Test for Navbar', () => {
  it('Check Navbar Disabled', () => {
    const tree = renderer
      .create(
        <Navbar
          data={datasetNavbarH}
          onChangeSelected={onCheckSelectedMock}
          typeOfNavbar="horizontal"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
