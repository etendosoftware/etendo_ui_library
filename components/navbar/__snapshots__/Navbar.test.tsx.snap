// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running Test for Navbar Check Navbar Disabled 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/etendo-erp.png",
            }
          }
          style={
            Object {
              "height": 28,
              "width": 112,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "width": 550,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/images/icons/binoculars.png",
              }
            }
            style={
              Object {
                "height": 28,
                "marginRight": "25px",
                "width": 28,
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/images/icons/video.png",
              }
            }
            style={
              Object {
                "height": 28,
                "marginRight": "25px",
                "width": 28,
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/images/icons/chat.png",
              }
            }
            style={
              Object {
                "height": 28,
                "marginRight": "25px",
                "width": 28,
              }
            }
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/images/icons/userNoBorder.png",
              }
            }
            style={
              Object {
                "height": 28,
                "marginRight": "25px",
                "width": 28,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
